src/gluonts/model/deeprenewal/_estimator.py

```diff
@@ -0,0 +1,378 @@
+        assert (
+            context_length is None or context_length > 0
+        ), "The value of `context_length` should be > 0"
+        assert num_layers > 0, "The value of `num_layers` should be > 0"
+        assert num_cells > 0, "The value of `num_cells` should be > 0"
+        assert dropout_rate >= 0, "The value of `dropout_rate` should be >= 0"
+        assert (cardinality is not None and use_feat_static_cat) or (
+            cardinality is None and not use_feat_static_cat
+        ), "You should set `cardinality` if and only if `use_feat_static_cat=True`"
+        assert cardinality is None or all(
+            [c > 0 for c in cardinality]
+        ), "Elements of `cardinality` should be > 0"
+        assert embedding_dimension is None or all(
+            [e > 0 for e in embedding_dimension]
+        ), "Elements of `embedding_dimension` should be > 0"
+        assert (
+            num_parallel_samples > 0
+        ), "The value of `num_parallel_samples` should be > 0"
+
+        self.freq = freq
+        self.context_length = (
+            context_length if context_length is not None else prediction_length
+        )
+        self.prediction_length = prediction_length
+        # self.distr_output = distr_output
+        self.distr_output_m = NegativeBinomialOutput()
+        self.distr_output_q = NegativeBinomialOutput()
+        self.distr_output_m.dtype = dtype
+        self.distr_output_q.dtype = dtype
+        self.num_layers = num_layers
+        self.num_cells = num_cells
+        self.cell_type = cell_type
+        self.dropout_rate = dropout_rate
+        self.use_feat_dynamic_real = use_feat_dynamic_real
+        self.use_feat_static_cat = use_feat_static_cat
+        self.use_feat_static_real = use_feat_static_real
+        self.cardinality = (
+            cardinality if cardinality and use_feat_static_cat else [1]
+        )
+        self.embedding_dimension = (
+            embedding_dimension
+            if embedding_dimension is not None
+            else [min(50, (cat + 1) // 2) for cat in self.cardinality]
+        )
+        self.scaling = scaling
+        self.lags_seq = (
+            lags_seq
+            if lags_seq is not None
+            else get_lags_for_frequency(freq_str=freq)
+        )
+        self.time_features = (
+            time_features
+            if time_features is not None
+            else time_features_from_frequency_str(self.freq)
+        )
+
+        self.history_length = self.context_length + max(self.lags_seq)
+
+        self.num_parallel_samples = num_parallel_samples
+        self.forecast_type = forecast_type
+
+    def create_transformation(self) -> Transformation:
+        remove_field_names = [FieldName.FEAT_DYNAMIC_CAT]
+        if not self.use_feat_static_real:
+            remove_field_names.append(FieldName.FEAT_STATIC_REAL)
+        if not self.use_feat_dynamic_real:
+            remove_field_names.append(FieldName.FEAT_DYNAMIC_REAL)
+
+        return Chain(
```

Comment:
Most of the other comments are still valid.. we can discuss and work on them..

Reply:
I feel we shud keep the ability to include static feat cat etc, but make them turned off  by default.. For some use cases, especially when we have lots of data, it can be beneficial. What do you think?

Reply:
I agree for static categorical, but dynamic features are hard to define for intermittent demand. For example, does one have a feature for issue points or for all time intervals (or another sparse arrival pattern altogether)? May I propose we exclude dynamic features?